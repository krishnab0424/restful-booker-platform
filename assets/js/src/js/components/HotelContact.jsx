import React from 'react';
import { API } from '../libs/Api';

export default class HotelContact extends React.Component {

    constructor() {
        super();

        this.state = {
            contact : {
                name : "",
                email : "",
                phone : "",
                subject : "",
                description : "",
            },
            submitted : false,
            errors : []
        }

        this.updateState = this.updateState.bind(this);
        this.submitForm = this.submitForm.bind(this);
    }

    updateState(event){
        let currentState = this.state.contact;

        currentState[event.target.id] = event.target.value;

        this.setState({contact : currentState});
    }

    submitForm(){
        API.postMessage(this);
    }

    render(){
        let form = '';
        let errors = '';

        if(this.state.errors.length > 0){
            errors = <div className="alert alert-danger" style={{marginBottom : 5 + "rem"}}>
                {this.state.errors.map((value) => {
                    return <p key={value}>{value}</p>
                })}
            </div>
        }

        if(this.state.submitted){
            form = <div style={{height : "412px"}}>
                    <h2>Thanks for getting in touch {this.state.contact.name}!</h2>
                    <p>We'll get back to you about</p>
                    <p style={{fontWeight : "bold"}}>{this.state.contact.subject}</p>
                    <p>as soon as possible.</p>
                    </div>
        } else {
            form = <form>
                    <div className="input-group mb-3">
                        <div className="input-group-prepend">
                            <span className="input-group-text" id="basic-addon1"><span className="fa fa-id-card"></span></span>
                        </div>
                        <input type="text" className="form-control" placeholder="Name" aria-label="Name" id="name" aria-describedby="basic-addon1" onChange={this.updateState} />
                    </div>
                    <div className="input-group mb-3">
                        <div className="input-group-prepend">
                            <span className="input-group-text" id="basic-addon1"><span className="fa fa-envelope"></span></span>
                        </div>
                        <input type="text" className="form-control" placeholder="Email" aria-label="Email" id="email" aria-describedby="basic-addon1" onChange={this.updateState} />
                    </div>
                    <div className="input-group mb-3">
                        <div className="input-group-prepend">
                            <span className="input-group-text" id="basic-addon1"><span className="fa fa-phone"></span></span>
                        </div>
                        <input type="text" className="form-control" placeholder="Phone" aria-label="Phone" id="phone" aria-describedby="basic-addon1" onChange={this.updateState} />
                    </div>
                    <div className="input-group mb-3">
                        <div className="input-group-prepend">
                            <span className="input-group-text" id="basic-addon1"><span className="fa fa-envelope"></span></span>
                        </div>
                        <input type="text" className="form-control" placeholder="Subject" aria-label="Subject" id="subject" aria-describedby="basic-addon1" onChange={this.updateState} />
                    </div>
                    <div className="input-group">
                        <div className="input-group-prepend">
                            <span className="input-group-text">Message</span>
                        </div>
                        <textarea className="form-control" aria-label="Description" id="description" rows="5" onChange={this.updateState}></textarea>
                    </div>
                    <br />
                    {errors}
                    <button type='button' className='btn btn-outline-primary float-right' id="submitContact" onClick={this.submitForm}>Submit</button>
                </form>
        }

        if(this.props.contact){
            return <div className='row contact'>
                        <div className='col-sm-1'></div>
                        <div className='col-sm-5'>
                            {form}
                        </div>
                        <div className='col-sm-5'>
                            <p><span className="fa fa-home"></span> {this.props.contact.name}</p>
                            <p><span></span> {this.props.contact.address}</p>
                            <p><span className="fa fa-phone"></span> {this.props.contact.phone}</p>
                            <p><span className="fa fa-envelope"></span> {this.props.contact.email}</p>
                        </div>
                        <div className='col-sm-1'></div>
                    </div>
        } else {
            return <p>ERROR: Missing contact details to render</p>
        }
    }

}