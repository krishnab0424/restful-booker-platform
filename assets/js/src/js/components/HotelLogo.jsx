import React from 'react';

export default class HotelLogo extends React.Component {

    render(){
        if(this.props.logoDetails){
            return <img src={this.props.logoDetails} className='hotel-logoUrl' alt='Hotel logoUrl' />
        } else {
            return <p>ERROR: Url link not provided</p>
        }
    }

}