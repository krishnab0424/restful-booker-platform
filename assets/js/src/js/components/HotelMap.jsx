import React from 'react';
import Map from 'pigeon-maps';
import Marker from 'pigeon-marker';

const getProvider = (x, y, z) => `https://cartodb-basemaps-a.global.ssl.fastly.net/light_all/${z}/${x}/${y}.png`;

export default class HotelMap extends React.Component {

  constructor(){
    super();
  }

  render() {
    if(this.props.mapDetails){
      const marker = [
        <Marker 
          key={`marker_${this.props.name}`} 
          anchor={[this.props.mapDetails.latitude, this.props.mapDetails.longitude]} 
          payload={this.props.name} />
      ]
  
      return (
        <div className="map">
          <Map
            width={window.innerWidth}
            height={600}
            defaultCenter={[this.props.mapDetails.latitude, this.props.mapDetails.longitude]}
            defaultZoom={17}
            provider={getProvider}
            mouseEvents={false}
          >
          
          {marker}
          </Map>
        </div>
      );
    } else {
      return <div className="map">
        <p>ERROR: Invalid or details missing for map to render</p>
      </div>
    }
  }
}
