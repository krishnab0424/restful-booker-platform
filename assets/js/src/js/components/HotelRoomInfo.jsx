import React from 'react';
import RoomBookingForm from './RoomBookingForm.jsx';

export default class HotelRoomInfo extends React.Component {

    constructor(){
        super();

        this.state = {
            book : false
        }

        this.toggleBooking = this.toggleBooking.bind(this);
    }

    toggleBooking(){
        if(this.state.book){
            this.setState({book : false})
        } else {
            this.setState({book : true})
        }
    }

    render(){
        let bookRoomView;
        let button;

        if(this.state.book){
            bookRoomView = <RoomBookingForm roomid={this.props.room.roomid} roomPrice={this.props.room.roomPrice} toggleBooking={this.toggleBooking} />
        } else {
            button = <button type='button' className='btn btn-outline-primary float-right openBooking' onClick={this.toggleBooking}>Book this room</button>
        }

        return(
            <div>
                <div className="row hotel-room-info">
                    <div className="col-sm-1"></div>
                    <div className="col-sm-3">
                        <img className="img-responsive hotel-img" src={this.props.room.image} alt={"Preview image of room" + this.props.room.roomNumber} />
                    </div>
                    <div className="col-sm-7">
                        {this.props.room.accessible && <span className="fa fa-wheelchair wheelchair"></span>}
                        <h3>{this.props.room.type}</h3>
                        <p>{this.props.room.description}</p>
                        <ul>
                            {this.props.room.features.map((feature, index) => {
                                return <li key={index}>{feature}</li>
                            })}
                        </ul>
                        {button}
                    </div>
                    <div className="col-sm-1"></div>
                </div>
                {bookRoomView}
            </div>
        )
    }

}