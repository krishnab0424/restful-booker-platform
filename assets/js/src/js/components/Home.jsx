import React from 'react';
import fetch from 'node-fetch';
import HotelRoomInfo from './HotelRoomInfo.jsx';
import HotelMap from './HotelMap.jsx';
import HotelLogo from './HotelLogo.jsx';
import HotelContact from './HotelContact.jsx';
import { API_ROOT } from '../api-config';

export default class Home extends React.Component {

    constructor() {
        super();

        this.state = {
            rooms : []
        }
    }

    componentDidMount() {
        fetch(API_ROOT + '/branding/', {
			method: 'GET',
			headers: {
				'Accept': 'application/json',
				'Content-Type': 'application/json'
			}
        })
        .then(res => res.json())
        .then(res => {
            this.setState(res);
        })
        .catch(e => console.log(e));

        fetch(API_ROOT + '/room/', {
            method: 'GET',
            headers: {
                'Accept': 'application/json',
                'Content-Type': 'application/json'
            }
        })
        .then(res => res.json())
        .then(res => {
            this.setState({rooms : res.rooms});
        })
        .catch(e => console.log(e));
    }

    render() {
        return (
            <div>
                <div className="container-fluid">
                    <div className="row">
                        <div className='col-sm-12 text-center'>
                            <HotelLogo logoDetails={this.state.logoUrl} />
                        </div>
                    </div>
                    <div className="row hotel-description">
                        <div className='col-sm-1'></div>
                        <div className='col-sm-10'>
                            <p>{this.state.description}</p>
                        </div>
                        <div className='col-sm-1'></div>
                    </div>
                    <div className="row room-header">
                        <div className='col-sm-1'></div>
                        <div className='col-sm-10'>
                            <h2>Rooms</h2>
                        </div>
                        <div className='col-sm-1'></div>
                    </div>
                    {this.state.rooms.map((roomDetails) => {
                        return <div key={roomDetails.roomid}><HotelRoomInfo room={roomDetails} /></div>
                    })}
                        <HotelContact contact={this.state.contact} />
                    <div className="row">
                        <div className='col-sm-12'>
                            <HotelMap name={this.state.name} mapDetails={this.state.map} />
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}
