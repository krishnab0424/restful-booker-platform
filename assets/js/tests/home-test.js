import React from 'react';
import Home from '../src/js/components/Home.jsx';
import HotelMap from '../src/js/components/HotelMap.jsx';
import HotelLogo from '../src/js/components/HotelLogo.jsx';
import HotelContact from '../src/js/components/HotelContact.jsx';
import HotelRoomInfo from '../src/js/components/HotelRoomInfo.jsx';
import nock from 'nock';

const homeState = {
    rooms : [{
        roomid: 1,
        roomNumber: 101,
        type: 'Standard Room',
        accessible: false,
        image : 'https://www.mwtestconsultancy.co.uk/img/room1.jpg',
        description: 'Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.',
        features: [
            'Internet/Wi-fi',
            'Jacuzzi Bathroom',
            'Air conditioning',
            'High Definition TV',
            'Mini-bar'
        ]
    },{
        roomid: 2,
        roomNumber: 102,
        type: 'Single',
        accessible: true,
        image: 'https://www.mwtestconsultancy.co.uk/img/room2.jpg',
        description: 'Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.',
        features: [
            'Internet/Wi-fi',
            'Air conditioning',
            'Mini-bar'
        ]
    }],
    name : 'Shady meadows B&B',
    map : {
        latitude : 52.6351204,
        longitude : 1.2733774
    },
    logoUrl : 'https://www.mwtestconsultancy.co.uk/img/rbp-logo.png',
    description : "Welcome to Shady Meadows, a delightful Bed & Breakfast nestled in the hills on Newfordburyshire. A place so beautiful you will never want to leave. All our rooms have comfortable beds and we provide breakfast from the locally sourced supermarket. It's a delightful place.",
    contact : {
        name : 'Shady Meadows B&B',
        address : 'The Old Farmhouse, Shady Street, Newfordburyshire, NE1 410S',
        phone : '0123456789',
        email : 'fake@fakeemail.com'
    }
}

nock('http://localhost')
    .get('/room/')
    .reply(200, homeState);

nock('http://localhost')
    .get('/branding/')
    .reply(200, {
        name: 'A new name',
        map: {
            latitude: 88.123,
            longitude: 11.123
        },
        logoUrl: 'https://www.mwtestconsultancy.co.uk/url/update.png',
        description: 'Updated description',
        contact: {
            name: 'Another B&B',
            address: 'Somewhere else',
            phone: '99999999999',
            email: 'another@fakeemail.com'
        }
    });

test('Home page renders', (done) => {
    const homeComponent = shallow(
        <Home />
    )

    setTimeout(() => {
        expect(homeComponent).toMatchSnapshot();
        done();
    }, 1000);
});

test('Room info for home page renders', () => {
    const roomDetails = {
        roomid: 1,
        roomNumber: 101,
        type: 'Standard Room',
        beds: 2,
        accessible: false,
        image : 'https://www.mwtestconsultancy.co.uk/img/room1.jpg',
        description: 'Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.',
        features: [
            'Internet/Wi-fi',
            'Jacuzzi Bathroom',
            'Air conditioning',
            'High Definition TV',
            'Mini-bar'
        ]
    }

    const hotelRoomInfoComponent = shallow(
        <HotelRoomInfo room={roomDetails} />
    )

    expect(hotelRoomInfoComponent).toMatchSnapshot();
});

test('Booking info for room renders when button selected', () => {
    const roomDetails = {
        roomid: 1,
        roomNumber: 101,
        type: 'Standard Room',
        beds: 2,
        accessible: false,
        image : 'https://www.mwtestconsultancy.co.uk/img/room1.jpg',
        description: 'Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.',
        features: [
            'Internet/Wi-fi',
            'Jacuzzi Bathroom',
            'Air conditioning',
            'High Definition TV',
            'Mini-bar'
        ]
    }

    const hotelRoomInfoComponent = shallow(
        <HotelRoomInfo room={roomDetails} />
    )

    hotelRoomInfoComponent.find('.openBooking').simulate('click');

    expect(hotelRoomInfoComponent).toMatchSnapshot();
});

test('Map for home page renders', () => {
    const mapDetails = {
        latitude : 52.6351204,
        longitude : 1.2733774
    }

    const mapComponent = shallow(
        <HotelMap name={'Shady Meadows'} mapDetails={mapDetails} />
    )

    expect(mapComponent).toMatchSnapshot();
});

test('Map for home page renders error', () => {
    const mapComponent = shallow(
        <HotelMap />
    )

    expect(mapComponent).toMatchSnapshot();
});

test('Logo for home page renders', () => {
    const logoDetails = {
        url : 'https://www.mwtestconsultancy.co.uk/img/rbp-logo.png'
    }

    const logoComponent = shallow(
        <HotelLogo logoDetails={logoDetails} />
    )

    expect(logoComponent).toMatchSnapshot();
});

test('Logo for home page errors render', () => {
    const logoComponent = shallow(
        <HotelLogo />
    )

    expect(logoComponent).toMatchSnapshot();
});

test('Contact form for home page renders', () => {
    const contactDetails = {
        name : 'Shady Meadows B&B',
        address : 'The Old Farmhouse, Shady Street, Newfordburyshire, NE1 410S',
        phone : '0123456789',
        email : 'fake@fakeemail.com'
    }

    const contactComponent = shallow(
        <HotelContact contact={contactDetails} />
    )

    expect(contactComponent).toMatchSnapshot();
});

test('Contact form errors render', () => {
    const contactComponent = shallow(
        <HotelContact />
    )

    expect(contactComponent).toMatchSnapshot();
});