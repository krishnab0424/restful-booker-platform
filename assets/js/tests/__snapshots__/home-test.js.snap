// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Booking info for room renders when button selected 1`] = `
<div>
  <div
    className="row hotel-room-info"
  >
    <div
      className="col-sm-1"
    />
    <div
      className="col-sm-3"
    >
      <img
        alt="Preview image of room101"
        className="img-responsive hotel-img"
        src="https://www.mwtestconsultancy.co.uk/img/room1.jpg"
      />
    </div>
    <div
      className="col-sm-7"
    >
      <h3>
        Standard Room
      </h3>
      <p>
        Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.
      </p>
      <ul>
        <li
          key="0"
        >
          Internet/Wi-fi
        </li>
        <li
          key="1"
        >
          Jacuzzi Bathroom
        </li>
        <li
          key="2"
        >
          Air conditioning
        </li>
        <li
          key="3"
        >
          High Definition TV
        </li>
        <li
          key="4"
        >
          Mini-bar
        </li>
      </ul>
    </div>
    <div
      className="col-sm-1"
    />
  </div>
  <RoomBookingForm
    roomid={1}
    toggleBooking={[Function]}
  />
</div>
`;

exports[`Contact form errors render 1`] = `
<p>
  ERROR: Missing contact details to render
</p>
`;

exports[`Contact form for home page renders 1`] = `
<div
  className="row contact"
>
  <div
    className="col-sm-1"
  />
  <div
    className="col-sm-5"
  >
    <form>
      <div
        className="input-group mb-3"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="basic-addon1"
          >
            <span
              className="fa fa-id-card"
            />
          </span>
        </div>
        <input
          aria-describedby="basic-addon1"
          aria-label="Name"
          className="form-control"
          id="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
        />
      </div>
      <div
        className="input-group mb-3"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="basic-addon1"
          >
            <span
              className="fa fa-envelope"
            />
          </span>
        </div>
        <input
          aria-describedby="basic-addon1"
          aria-label="Email"
          className="form-control"
          id="email"
          onChange={[Function]}
          placeholder="Email"
          type="text"
        />
      </div>
      <div
        className="input-group mb-3"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="basic-addon1"
          >
            <span
              className="fa fa-phone"
            />
          </span>
        </div>
        <input
          aria-describedby="basic-addon1"
          aria-label="Phone"
          className="form-control"
          id="phone"
          onChange={[Function]}
          placeholder="Phone"
          type="text"
        />
      </div>
      <div
        className="input-group mb-3"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="basic-addon1"
          >
            <span
              className="fa fa-envelope"
            />
          </span>
        </div>
        <input
          aria-describedby="basic-addon1"
          aria-label="Subject"
          className="form-control"
          id="subject"
          onChange={[Function]}
          placeholder="Subject"
          type="text"
        />
      </div>
      <div
        className="input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
          >
            Message
          </span>
        </div>
        <textarea
          aria-label="Description"
          className="form-control"
          id="description"
          onChange={[Function]}
          rows="5"
        />
      </div>
      <br />
      <button
        className="btn btn-outline-primary float-right"
        id="submitContact"
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </form>
  </div>
  <div
    className="col-sm-5"
  >
    <p>
      <span
        className="fa fa-home"
      />
       
      Shady Meadows B&B
    </p>
    <p>
      <span />
       
      The Old Farmhouse, Shady Street, Newfordburyshire, NE1 410S
    </p>
    <p>
      <span
        className="fa fa-phone"
      />
       
      0123456789
    </p>
    <p>
      <span
        className="fa fa-envelope"
      />
       
      fake@fakeemail.com
    </p>
  </div>
  <div
    className="col-sm-1"
  />
</div>
`;

exports[`Home page renders 1`] = `
<div>
  <div
    className="container-fluid"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-12 text-center"
      >
        <HotelLogo
          logoDetails="https://www.mwtestconsultancy.co.uk/url/update.png"
        />
      </div>
    </div>
    <div
      className="row hotel-description"
    >
      <div
        className="col-sm-1"
      />
      <div
        className="col-sm-10"
      >
        <p>
          Updated description
        </p>
      </div>
      <div
        className="col-sm-1"
      />
    </div>
    <div
      className="row room-header"
    >
      <div
        className="col-sm-1"
      />
      <div
        className="col-sm-10"
      >
        <h2>
          Rooms
        </h2>
      </div>
      <div
        className="col-sm-1"
      />
    </div>
    <div
      key="1"
    >
      <HotelRoomInfo
        room={
          Object {
            "accessible": false,
            "description": "Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.",
            "features": Array [
              "Internet/Wi-fi",
              "Jacuzzi Bathroom",
              "Air conditioning",
              "High Definition TV",
              "Mini-bar",
            ],
            "image": "https://www.mwtestconsultancy.co.uk/img/room1.jpg",
            "roomNumber": 101,
            "roomid": 1,
            "type": "Standard Room",
          }
        }
      />
    </div>
    <div
      key="2"
    >
      <HotelRoomInfo
        room={
          Object {
            "accessible": true,
            "description": "Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.",
            "features": Array [
              "Internet/Wi-fi",
              "Air conditioning",
              "Mini-bar",
            ],
            "image": "https://www.mwtestconsultancy.co.uk/img/room2.jpg",
            "roomNumber": 102,
            "roomid": 2,
            "type": "Single",
          }
        }
      />
    </div>
    <HotelContact
      contact={
        Object {
          "address": "Somewhere else",
          "email": "another@fakeemail.com",
          "name": "Another B&B",
          "phone": "99999999999",
        }
      }
    />
    <div
      className="row"
    >
      <div
        className="col-sm-12"
      >
        <HotelMap
          mapDetails={
            Object {
              "latitude": 88.123,
              "longitude": 11.123,
            }
          }
          name="A new name"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Logo for home page errors render 1`] = `
<p>
  ERROR: Url link not provided
</p>
`;

exports[`Logo for home page renders 1`] = `
<img
  alt="Hotel logoUrl"
  className="hotel-logoUrl"
  src={
    Object {
      "url": "https://www.mwtestconsultancy.co.uk/img/rbp-logo.png",
    }
  }
/>
`;

exports[`Map for home page renders 1`] = `
<div
  className="map"
>
  <Map
    animate={true}
    animateMaxScreens={5}
    defaultCenter={
      Array [
        52.6351204,
        1.2733774,
      ]
    }
    defaultZoom={17}
    dprs={Array []}
    height={600}
    limitBounds="center"
    maxZoom={18}
    metaWheelZoom={false}
    metaWheelZoomWarning="Use META+wheel to zoom!"
    minZoom={1}
    mouseEvents={false}
    provider={[Function]}
    touchEvents={true}
    twoFingerDrag={false}
    twoFingerDragWarning="Use two fingers to move the map"
    warningZIndex={100}
    width={1024}
    zoomSnap={true}
  >
    <Marker
      anchor={
        Array [
          52.6351204,
          1.2733774,
        ]
      }
      key="marker_Shady Meadows"
      payload="Shady Meadows"
    />
  </Map>
</div>
`;

exports[`Map for home page renders error 1`] = `
<div
  className="map"
>
  <p>
    ERROR: Invalid or details missing for map to render
  </p>
</div>
`;

exports[`Room info for home page renders 1`] = `
<div>
  <div
    className="row hotel-room-info"
  >
    <div
      className="col-sm-1"
    />
    <div
      className="col-sm-3"
    >
      <img
        alt="Preview image of room101"
        className="img-responsive hotel-img"
        src="https://www.mwtestconsultancy.co.uk/img/room1.jpg"
      />
    </div>
    <div
      className="col-sm-7"
    >
      <h3>
        Standard Room
      </h3>
      <p>
        Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.
      </p>
      <ul>
        <li
          key="0"
        >
          Internet/Wi-fi
        </li>
        <li
          key="1"
        >
          Jacuzzi Bathroom
        </li>
        <li
          key="2"
        >
          Air conditioning
        </li>
        <li
          key="3"
        >
          High Definition TV
        </li>
        <li
          key="4"
        >
          Mini-bar
        </li>
      </ul>
      <button
        className="btn btn-outline-primary float-right openBooking"
        onClick={[Function]}
        type="button"
      >
        Book this room
      </button>
    </div>
    <div
      className="col-sm-1"
    />
  </div>
</div>
`;
